pytorch_lightning/metrics/functional/iou.py

```diff
@@ -0,0 +1,101 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import torch
+from typing import Optional
+from pytorch_lightning.metrics.functional.classification import get_num_classes
+from pytorch_lightning.metrics.functional.reduction import reduce
+from pytorch_lightning.metrics.functional.confusion_matrix import _confusion_matrix_update
+
+
+def _iou_from_confmat(
+        confmat: torch.Tensor,
+        num_classes: int,
+        ignore_index: Optional[int] = None,
+        absent_score: float = 0.0,
+        reduction: str = 'elementwise_mean',
+):
+    intersection = torch.diag(confmat)
+    union = confmat.sum(0) + confmat.sum(1) - intersection
+
+    # If this class is absent in both target AND pred (union == 0), then use the absent_score for this class.
+    scores = intersection.float() / union.float()
+    scores[union == 0] = absent_score
+
+    # Remove the ignored class index from the scores.
+    if ignore_index is not None and ignore_index >= 0 and ignore_index < num_classes:
+        scores = torch.cat([
+            scores[:ignore_index],
+            scores[ignore_index + 1:],
+        ])
+    return reduce(scores, reduction=reduction)
```

Comment:
Does this kind of reduction makes sense for this metric or should we instead use the `class_reduce` function for better matching the `jaccard_score` from sklearn?
https://github.com/PyTorchLightning/pytorch-lightning/blob/eb9cb3c8017b1f207ac1d2c7168327fbe0b63cb5/pytorch_lightning/metrics/utils.py#L253-L291

Reply:
I think either is fine. I prefer to match `jaccard_score`. I can change the code if you like, do I want to keep `reduction` for backward compatibility?

Reply:
okay for now lets keep it as it is, then if you want to send a new PR which deprecates the `reduce` arg and introduce a new `average` arg.